### PR TITLE
7.2.1: Fixes and p8-platform removal

### PR DESCRIPTION
**Pull request by ksooo**
- Fix 'pvr.hts' appearing twice in every addon log message
- Fix calculation of start time, stop time and file size for multi-file recordings
- Remove p8-platform build dependency

### CODE REVIEW

File: src/tvheadend/utilities/TCPSocket.h [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    auto endpoints = asio::ip::tcp::resolver(m_context).resolve(m_host, std::to_string(m_port));
+
+    std::error_code error;
+    asio::async_connect(
+        m_socket, endpoints,
+        [&](const std::error_code& result_error, const asio::ip::tcp::endpoint&) {
+          error = result_error;
+        });
+
+    if (error)
```
Comment (AlwinEsch) [inline src/tvheadend/utilities/TCPSocket.h]: About here, as I understand his code of ASIO correct is the error not set until his work was done where is in independent thread, the `RunFor()` then only blocks until he is finished or timeout.

**Review by AlwinEsch — COMMENTED** [review]
Except the error check it looks good to me :+1:

File: src/tvheadend/utilities/TCPSocket.h (hunk 1 above)
Comment (ksooo) [reply src/tvheadend/utilities/TCPSocket.h]: Sorry, I do not understand what you mean.

Comment (ksooo) [reply src/tvheadend/utilities/TCPSocket.h]: Fixed.

File: CMakeLists.txt [hunk 2]
```diff
@@ -98,7 +96,7 @@ list(APPEND HTS_SOURCES
 
 add_subdirectory(lib/libhts)
 
-set(DEPLIBS ${p8-platform_LIBRARIES} hts)
+set(DEPLIBS hts)
 if(WIN32)
   list(APPEND DEPLIBS ws2_32)
```
Comment (AlwinEsch) [inline CMakeLists.txt]: This also no more needed as the `ASIO` makes `#   pragma comment(lib, "ws2_32.lib")`

Comment (ksooo) [reply CMakeLists.txt]: Okay, will remove, then.

**Comment by ksooo** [comment]
Please note that pvr.hts now uses kissnet socket implementation (https://github.com/Ybalrid/kissnet) , which requires c++ 17.

**Comment by phunkyfish** [comment]
Nice!

**Comment by ksooo** [comment]
@AlwinEsch any idea why Travis build still fails? Errors indicate that the compiler does not understand C++17...

**Comment by lrusak** [comment]
@ksooo the ubuntu version is too old, gcc 5.x doesn't support c++17. You need at least gcc 7 (ubuntu 18.04 bionic beaver)

**Pull request merged**